src/network/networkd-radv.c

```diff
@@ -658,19 +667,24 @@ int config_parse_radv_dns(
                 if (r == 0)
                         break;
 
-                if (in_addr_from_string(AF_INET6, w, &a) >= 0) {
-                        struct in6_addr *m;
-
-                        m = reallocarray(n->router_dns, n->n_router_dns + 1, sizeof(struct in6_addr));
-                        if (!m)
-                                return log_oom();
+                if (streq(w, "linklocal"))
+                        a = IN_ADDR_NULL;
+                else {
+                        r = in_addr_from_string(AF_INET6, w, &a);
+                        if (r < 0) {
+                                log_syntax(unit, LOG_ERR, filename, line, 0,
```

Comment:
hmm, while we are at it, the 5th argument should really be `r` here, i.e. the error code seen, not `0`

Also, I think it would make sense to refuse all-zero addresses here now, given that this is used internally to denote something special now.